lib/rmt/cli/repos_custom.rb

```diff
@@ -0,0 +1,119 @@
+class RMT::CLI::ReposCustom < RMT::CLI::Base
+
+  include ::RMT::CLI::ArrayPrintable
+
+  desc 'add URL NAME', 'Creates a custom repository.'
+  def add(url, name)
+    url += '/' unless url.end_with?('/')
+
+    previous_repository = Repository.find_by(external_url: url)
+
+    if previous_repository
+      raise RMT::CLI::Error.new('A repository by this URL already exists.')
+    end
+
+    repository_service.create_repository!(nil, url, {
+      name: name,
+      mirroring_enabled: 1,
+      autorefresh: 1,
+      enabled: 0
```

Comment:
It would be better to use the correct data types: `true`/`false` instead of `0`/`1`.